### PR TITLE
feat(wallet): add general transactions view

### PR DESCRIPTION
**Pull request by raducristianpopa**
### Context

- context of #259 

### Changes
- adds a general transactions view
- removes the single transactions page view (per payment pointer)
- pagination
- filtering

### CODE REVIEW

File: packages/wallet/frontend/src/lib/api/paymentPointer.ts [hunk 1]
```diff
@@ -174,6 +142,21 @@ const createPaymentPointerService = (): PaymentPointerService => ({
...
+      const response = await httpClient
+        .get('payment-pointers', {
+          headers: {
+            ...(cookies ? { Cookie: cookies } : {})
+          }
+        })
+        .json<ListPaymentPointerResult>()
+      return response
+    } catch (error) {
+      return getError(error, 'Unable to fetch payment pointers')
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/lib/api/paymentPointer.ts:156]:
```suggestion
      return getError(error, 'Unable to fetch payment pointers.')
```

File: packages/wallet/frontend/src/lib/hooks/useHttp.ts [hunk 2]
```diff
@@ -3,25 +3,33 @@ import { ErrorResponse, SuccessResponse } from '../httpClient'
...
+      fn: (a?: TArgs) => Promise<TResponse>,
       args: TArgs
     ) => {
+      setError(undefined)
       setIsLoading(true)
       const response = await fn(args)
+      if (!response?.success) {
+        setError(
+          response?.message ??
+            'We encountered a network error. Please try again!'
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/lib/hooks/useHttp.ts:24]:
```suggestion
            'We encountered a network error. Please try again.'
```

File: packages/wallet/frontend/src/pages/transactions.tsx [hunk 3]
```diff
@@ -0,0 +1,356 @@
...
+            label="Payment Pointer"
+            placeholder="Select payment pointer..."
+            value={
+              currentPaymentPointer.accountId !== currentAccount.value &&
+              currentAccount.value !== ''
+                ? { ...defaultOption, accountId: '' }
+                : currentPaymentPointer
+            }
+            onChange={(option) => {
+              console.log(
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/transactions.tsx:151]: I don't think we need console logs

File: packages/wallet/frontend/src/pages/transactions.tsx [hunk 4]
```diff
@@ -0,0 +1,356 @@
...
+                currentAccount.value &&
+                  currentPaymentPointer.accountId !== currentAccount.value
+              )
+              console.log(currentAccount.value, option?.accountId)
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/transactions.tsx:155]: ditto

File: packages/wallet/frontend/src/pages/transactions.tsx [hunk 5]
```diff
@@ -0,0 +1,356 @@
...
+                        intent={getStatusBadgeIntent(trx.status)}
+                        size="md"
+                        text={trx.status}
+                      />
+                    </Table.Cell>
+                    <Table.Cell>
+                      {
+                        formatAmount({
+                          value: trx.value ?? 0,
+
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/transactions.tsx:254]: no need for extra empty rows

File: packages/wallet/frontend/src/pages/transactions.tsx [hunk 6]
```diff
@@ -0,0 +1,356 @@
...
+          >
+            Previous
+          </Button>
+          <Button
+            className="disabled:pointer-events-none disabled:from-gray-400 disabled:to-gray-500"
+            aria-label="go to next page"
+            disabled={Number(pagination.page) + 1 > totalPages - 1}
+            onClick={() => {
+              const nextPage = Number(pagination.page) + 1
+              console.log(nextPage, totalPages - 1)
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/transactions.tsx:294]: console log

File: packages/wallet/frontend/src/pages/transactions.tsx (hunk 5 above)
Comment (raducristianpopa) [reply packages/wallet/frontend/src/pages/transactions.tsx:254]: Can you elaborate?

Comment (Tymmmy) [reply packages/wallet/frontend/src/pages/transactions.tsx:254]: I see empty rows after every object key/value

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**